src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    public static Figure getRandomFigure() {
+        Random random = new Random();
+        Colors color = ColorSupplier.getRandomColor();
```

Comment:
create Random and ColorSupplier as class-level fields - to avoid creating a new instance each time his method is invoked